database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/source/DBSource.java

```diff
@@ -66,6 +85,135 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     dbManager.validateJDBCPluginPipeline(pipelineConfigurer, getJDBCPluginId());
   }
 
+  class ListTablesRequest {
+    public String connectionString;
+    @Nullable
+    public String user;
+    @Nullable
+    public String password;
+    public String jdbcPluginName;
+    @Nullable
+    public String jdbcPluginType = "jdbc";
+  }
+
+  class GetSchemaRequest extends ListTablesRequest {
+    public String importQuery;
+  }
+
+  /**
+   * Endpoint method to get the list of tables in a database
+   * @param request {@link ListTablesRequest} containing information required for connection
+   * @param pluginContext context to create plugins
+   * @return list of table names
+   * @throws SQLException
+   * @throws InstantiationException
+   * @throws IllegalAccessException
+   * @throws BadRequestException If executing get tables threw {@link SQLSyntaxErrorException}
+   * we get the message and throw it as BadRequestException
+   */
+  @Path("listTables")
+  public List<String> listTables(ListTablesRequest request,
+                                 EndpointPluginContext pluginContext) throws SQLException,
+    InstantiationException, IllegalAccessException, BadRequestException {
+
+    JDBCDriverShim driverShim = loadPluginClassAndGetDriver(request, pluginContext);
+    List<String> tableNames = new ArrayList();
+
+    try {
+      // initialize connection
+      try (Connection connection = getConnection(request.connectionString, request.user, request.password)) {
+        DatabaseMetaData metadata = connection.getMetaData();
+        try (ResultSet rs = metadata.getTables(null, null, "%", null)) {
```

Comment:
do we get the table schemas back?  Would be nice to be able to know a table's schema as well as its name.